File: VsCodeC++/projects/PredictingCollisions/PredictingCollisions.java
import java.io.*;
import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.*;

public class PredictingCollisions {

  static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
  static StringTokenizer st;
  static int fun[] = new int[6];
  static int der[] = new int[6];
  public static void main(String[] args) throws IOException {
    int n = readInt();
    for(int i=n; i>=0; i--){
        fun[i]=readInt();
    }
    int m = readInt();
    for(int i=m; i>=0; i--){
        fun[i]-=readInt();
    }
    for(int i=1; i<6; i++){
        der[i-1]=fun[i]*i;
    }
    double a = (double)readInt();
    double b = (double)readInt();
    if(a==0)a=b;
    BigDecimal c = new BigDecimal(a);
    while(calcFun(c).abs().compareTo(new BigDecimal("0.00000001"))==1){
        c=c.subtract(calcFun(c).divide(calcDer(c),20,RoundingMode.CEILING));
    }
    System.out.println(c.doubleValue());
  }
  static BigDecimal calcFun(BigDecimal a){
    BigDecimal ret = new BigDecimal("0");
    for(int i=0; i<6; i++){
        ret = ret.add(a.pow(i).multiply(new BigDecimal(fun[i])));
    }
    return ret;
  }
  static BigDecimal calcDer(BigDecimal a){
    BigDecimal ret = new BigDecimal("0");
    for(int i=0; i<6; i++){
        ret = ret.add(a.pow(i).multiply(new BigDecimal(der[i])));
    }
    return ret;
  }
  static String next() throws IOException {
    while (st == null || !st.hasMoreTokens())
      st = new StringTokenizer(br.readLine().trim());
    return st.nextToken();
  }

  static long readLong() throws IOException {
    return Long.parseLong(next());
  }

  static int readInt() throws IOException {
    return Integer.parseInt(next());
  }

  static double readDouble() throws IOException {
    return Double.parseDouble(next());
  }

  static char readCharacter() throws IOException {
    return next().charAt(0);
  }

  static String readLine() throws IOException {
    return br.readLine().trim();
  }
}

File: DMOJ&MCPT/PredictingCollisions/PredictingCollisions.cpp
#include <bits/stdc++.h>

using namespace std;
int func[6];
double calcFun(double n){
    double ret=0;
    for(int i=0; i<6; i++){
        ret+=func[i]*pow(n,i);
    }
    return ret;
}
int main() {
    int n,m;
    scanf("%d",&n);
    for(int i=n; i>=0; i--){
        scanf("%d",&func[i]);
    }
    scanf("%d",&m);
    for(int i=m,a; i>=0; i--){
        scanf("%d",&a);
        func[i]-=a;
    }
    double a,b;
    scanf("%lf %lf",&a,&b);
    double c=(a+b)/2;
    while (b-a>0.000000001){
        c = (a+b)/2;
        if(calcFun(c)*calcFun(a)>0){
            a=c;
        }else{
            b=c;
        }
    }
    printf("%.8f",c);
    return 0;
}